### PR TITLE
Flip canvas n stuff

### PR DESCRIPTION
**Pull request by flabbet**
- [x] Flip whole canvas vertically and horizontally
- [x] Flip individual layers along tight bounds center
- [x] Fixed palettes context menu
- [x] Rotate canvas
- [x] Rotate layers
- [x] Fix palette browser non-refreshing after importing palette from file

### CODE REVIEW

**Review by CPKreuz — CHANGES_REQUESTED** [review]
image

Maybe put the flip and the rotate in seperate menus

image

Some icons would be cool

File: src/PixiEditor/Views/MainWindow.xaml [hunk 1]
```diff
@@ -256,21 +256,23 @@
...
-                    </MenuItem>
-                    <Separator/>
-                    <MenuItem Header="_Flip Horizontal" Command="{Binding DocumentSubViewModel.FlipCommand}" CommandParameter="Horizontal"/>
-                    <MenuItem Header="_Flip Vertical" Command="{Binding DocumentSubViewModel.FlipCommand}" CommandParameter="Vertical"/>
-                -->
+                        <Separator/>
+                        <MenuItem Header="_Rotation">
+                            <MenuItem Header="Rotate Image 90&#186;" cmds:Menu.Command="PixiEditor.Document.Rotate90Deg"/>
+                            <MenuItem Header="Rotate Image 180&#186;" cmds:Menu.Command="PixiEditor.Document.Rotate180Deg"/>
+                            <MenuItem Header="Rotate Image 270&#186;" cmds:Menu.Command="PixiEditor.Document.Rotate270Deg"/>
```
Comment (CPKreuz) [inline src/PixiEditor/Views/MainWindow.xaml:263]: I feel like 270° should be written as -90° or using counter-clockwise notations

File: src/PixiEditor/Views/MainWindow.xaml [hunk 2]
```diff
@@ -256,21 +256,23 @@
...
+                            
+                            <Separator/>
+                            <MenuItem Header="Rotate Selected Layers 90&#186;" cmds:Menu.Command="PixiEditor.Document.Rotate90DegLayers"/>
+                            <MenuItem Header="Rotate Selected Layers 180&#186;" cmds:Menu.Command="PixiEditor.Document.Rotate180DegLayers"/>
+                            <MenuItem Header="Rotate Selected Layers 270&#186;" cmds:Menu.Command="PixiEditor.Document.Rotate270DegLayers"/>
+                        </MenuItem>
+                        <MenuItem Header="_Flipping">
```
Comment (CPKreuz) [inline src/PixiEditor/Views/MainWindow.xaml:270]: Maybe just `Flip` fit's more instead of `Flipping`

**Review by CPKreuz — CHANGES_REQUESTED** [review]
image

Half of the image gets caught of after rotating the image by 90° twice

Happens with images that have a non 1:1 aspect ratio

**Comment by flabbet** [comment]
Nice catch. Fixed, please test again. I noticed masks are rotating weirdly, so it's worth keeping in mind

File: src/PixiEditor/Views/MainWindow.xaml [hunk 3]
```diff
@@ -575,9 +577,16 @@
...
 																		Header="_Paste"
 																		cmds:ContextMenu.Command="PixiEditor.Clipboard.Paste" />
+                                                                            <Separator />
+                                                                            <MenuItem Header="Flip _Horizontally" cmds:Menu.Command="PixiEditor.Document.FlipLayersHorizontal"/>
+                                                                            <MenuItem Header="Flip _Vertically" cmds:Menu.Command="PixiEditor.Document.FlipLayersVertical"/>
+                                                                            <Separator />
+                                                                            <MenuItem Header="Rotate Right" cmds:Menu.Command="PixiEditor.Document.Rotate90DegLayers"/>
+                                                                            <MenuItem Header="Rotate 180&#186;" cmds:Menu.Command="PixiEditor.Document.Rotate180DegLayers"/>
+                                                                            <MenuItem Header="Rotate Left" cmds:Menu.Command="PixiEditor.Document.Rotate270DegLayers"/>
```
Comment (CPKreuz) [inline src/PixiEditor/Views/MainWindow.xaml:586]:
```suggestion
                                                                            <MenuItem Header="Rotate 90" cmds:Menu.Command="PixiEditor.Document.Rotate90DegLayers"/>
                                                                            <MenuItem Header="Rotate 180&#186;" cmds:Menu.Command="PixiEditor.Document.Rotate180DegLayers"/>
                                                                            <MenuItem Header="Rotate -90" cmds:Menu.Command="PixiEditor.Document.Rotate270DegLayers"/>
```

File: src/PixiEditor/ViewModels/SubViewModels/Document/DocumentManagerViewModel.cs [hunk 4]
```diff
@@ -55,6 +57,87 @@ public void ClipCanvas()
...
+    }
+    
+    [Command.Basic("PixiEditor.Document.Rotate270DegLayers", "Rotate Selected Layers 270 deg", "Rotate Selected Layers 270 deg", CanExecute = "PixiEditor.HasDocument")]
+    public void Rotate270DegLayers()
+    {
+        if (ActiveDocument?.SelectedStructureMember == null)
+            return;
+        
+        ActiveDocument?.Operations.RotateImage(RotationAngle.D270, ActiveDocument.GetSelectedMembers());
+    }
```
Comment (CPKreuz) [inline src/PixiEditor/ViewModels/SubViewModels/Document/DocumentManagerViewModel.cs:140]: This should all be doable using two methods using for whole image and for layers using a parameter

**Review by CPKreuz — APPROVED** [review]
Except for those inconsistencies in the context menu and the usage of the command attributes it looks good

**Pull request merged**